cli/popper/commands/cmd_add.py

```diff
@@ -70,24 +76,29 @@ def cli(ctx, pipeline, folder, branch):
     )
 
     # Downloading and extracting the tarfile
-    with tarfile.open(
-            mode='r:gz', fileobj=BytesIO(r.content)) as t:
+    with tarfile.open(mode='r:gz', fileobj=BytesIO(r.content)) as t:
         t.extractall()
 
-    os.rename('{}-{}/pipelines/{}'.format(
-        repo, branch, pipe_name), os.path.join(folder, pipe_name))
-    shutil.rmtree('{}-{}'.format(repo, branch))
+    try:
+        os.rename('{}-{}/pipelines/{}'.format(
+            repo, branch, pipe_name), pipelines_dir)
+    except OSError:
```

Comment:
is the folder not being empty the _only_ cause of an `OSError`? Maybe we should explicitly check whether the given folder is not empty. In general, exceptions are usually used to check for unexpected behavior, as opposed to control the flow of a program. A common practice is to check what the code is expecting when the exception is thrown instead. In this case, that the path is not empty (see comment above)

Reply:
Yes, we should do both.